src/fork/pty/master/ptsname_r_macos.rs

```diff
@@ -0,0 +1,133 @@
+//! macOS implementation of ptsname_r
+//!
+//! As `ptsname_r()` is not available on macOS, this provides a compatible
+//! implementation using the `TIOCPTYGNAME` ioctl syscall.
+//!
+//! Based on: https://tarq.net/posts/ptsname-on-osx-with-rust/
+
+#[cfg(any(target_os = "macos"))]
+pub unsafe fn ptsname_r(
+    fd: libc::c_int,
+    buf: *mut libc::c_char,
+    buflen: libc::size_t,
+) -> libc::c_int {
+    const IOCTL_BUF_SIZE: usize = 128;
+
+    if buf.is_null() || buflen == 0 {
+        return libc::EINVAL;
+    }
+
+    let mut ioctl_buf: [libc::c_char; IOCTL_BUF_SIZE] = [0; IOCTL_BUF_SIZE];
+
+    if libc::ioctl(fd, libc::TIOCPTYGNAME as libc::c_ulong, &mut ioctl_buf) != 0 {
+        return *libc::__error();
+    }
+
+    let mut len = 0;
```

Comment:
Totally fine to leave it like this, but for this sort of null-seaking, `memchr` (either the pure rust crate https://crates.io/crates/memchr, or the libc function https://docs.rs/libc/0.2.177/libc/fn.memchr.html) will do this a lot faster because they take advantage of SIMD hardware acceleration. It definitely won't matter for performance since the strings are so small here, so this is mostly just me being nerd sniped.

Reply:
TIL thank you. To not bring a new dependency, went for the libc one; https://github.com/shell-pool/shpool_pty/pull/4/commits/3bca7173b1c3d93a5abf7d916ac5448c0d63eb78